src/NodeCompiler/Exception/UnableToCompileNode.php

```diff
@@ -5,7 +5,56 @@
 namespace Roave\BetterReflection\NodeCompiler\Exception;
 
 use LogicException;
+use PhpParser\Node;
+use Roave\BetterReflection\NodeCompiler\CompilerContext;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use function get_class;
+use function sprintf;
 
 class UnableToCompileNode extends LogicException
 {
+    public static function forUnRecognizedExpressionInContext(Node\Expr $expression, CompilerContext $context) : self
+    {
+        return new self(sprintf(
+            'Unable to compile expression in %s: unrecognized node type %s at line %d',
+            self::compilerContextToContextDescription($context),
+            get_class($expression),
+            $expression->getLine()
+        ));
+    }
+
+    public static function becauseOfNotFoundClassConstantReference(
+        CompilerContext $fetchContext,
+        ReflectionClass $targetClass,
+        Node\Expr\ClassConstFetch $constantFetch
+    ) : self {
+        /** @var Node\Identifier $constantFetch ->name */
+        return new self(sprintf(
+            'Could not locate constant %s::%s while trying to evaluate constant expression in %s at line %s',
+            $targetClass->getName(),
+            $constantFetch->name->name,
+            self::compilerContextToContextDescription($fetchContext),
+            $constantFetch->getLine()
+        ));
+    }
+
+    public static function becauseOfOfNotFoundConstantReference(
```

Comment:
Is this the best name for this method, it is pretty similar to "becauseOfNotFoundClassConstantReference"

Reply:
Constant and ClassConstant are different concepts both for this library and for PHP-Parser

Reply:
never mind then 👼